spec/chore/worker_spec.rb

```diff
@@ -2,6 +2,28 @@
 
 describe Chore::Worker do
 
+  class FakeDalli
+    def initialize
+      @store = {}
+    end
+    def add(id, val, ttl)
+      if @store[id]
+        return false
+      else
+        @store[id] = val
+        Thread.new do
+          sleep ttl
+          @store.delete(id)
+        end
```

Comment:
Using threads in specs like this can lead to unpredictable results / slow spaces.  In fakememcached, we store the ttl along with the value and check that when looking up the value.  I would suggest a similar approach here.